### PR TITLE
feat(hermes): ingest all profile databases

### LINKED ISSUE #79: [feature]: Ingest stats from all Hermes profiles
### What problem does this solve?

Currently only the main Hermes profile is included in the analysis.

### Proposed solution

Find all `profiles/*/state.db` within the Hermes directory, keep one entry in the state for each of them and collect them all.

### Alternatives considered

_No response_

### Category

CLI command

### Anything else?

Working on a PR for it

### PR DESCRIPTION
**Pull request by Fryuni**
## Summary

Adds Hermes multi-profile ingestion so Token Tracker reads both the default `~/.hermes/state.db` and every discovered `~/.hermes/profiles/*/state.db`. This fixes missing Hermes usage from non-default profiles while keeping per-profile cursors isolated to avoid session-id collision deltas.

Closes #79.

## Changes

- Discover Hermes profile databases under `profiles/<name>/state.db` alongside the default database.
- Store named profile cursors under `cursors.hermes.profiles[profileName]` while preserving the existing top-level default-profile cursor fields for compatibility.
- Update sync to pass the Hermes base directory and add regression coverage for multi-profile ingestion, profile-only installs, and isolated snapshots.
- Document the Hermes multi-profile ingestion change in the changelog.

## Testing

- `node --test --test-name-pattern=Hermes test/rollout-parser.test.js` — passes (8/8)
- `node --test test/rollout-parser.test.js` — observed two pre-existing unrelated failures in OpenCode rewrite and Copilot rotation tests; Hermes coverage passed during this full run.

## Summary by CodeRabbit

* **New Features**
  * Passive token tracking now discovers and aggregates tokens across the default and per-profile Hermes databases, maintaining independent cursor state per profile (works when default DB is absent).

* **Bug Fixes**
  * More accurate conversation counts, safer cursor advancement to avoid missing/double-counting, and handling of same-second inserts so incremental parses don’t drop recent rows; finalized-session deltas now enqueue correct totals.

* **Tests**
  * Added integration tests for multi-profile discovery, per-profile cursor isolation, same-second inserts, and finalized-session delta handling.

[Review Change Stack](https://app.coderabbit.ai/change-stack/mm7894215/TokenTracker/pull/85?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/lib/rollout.js [hunk 1]
```diff
@@ -3005,18 +3005,57 @@ async function parseKiroIncremental({ dbPath, jsonlPath, cursors, queuePath, onP
...
+    ? [...new Set(unfinishedSessionIds.filter((id) => typeof id === "string" && id.length > 0))]
+    : [];
+  const forceIncludeSql = forceIds.length > 0
+    ? ` OR id IN (${forceIds.map(sqliteStringLiteral).join(",")})`
+    : "";
+  // Fetch sessions that started at/after the cursor, sessions that are still
+  // in-progress (ended_at IS NULL), OR sessions that were previously observed
+  // unfinished.  Hermes updates token counts in real-time, including a final
+  // delta when an active session later gets ended_at set.
+  const sql = `SELECT id, model, started_at, ended_at, input_tokens, output_tokens, cache_read_tokens, cache_write_tokens, reasoning_tokens, message_count FROM sessions WHERE (started_at >= ${since} OR ended_at IS NULL${forceIncludeSql}) AND (input_tokens > 0 OR output_tokens > 0 OR cache_read_tokens > 0 OR reasoning_tokens > 0) ORDER BY started_at ASC`;
```
[the remark this answers is not in the document]
Comment (Fryuni) [reply src/lib/rollout.js:3058]: Hermes (and most providers) doesn't support requests that just write to the cache for future use. The cache is written on the request for output tokens.

**Comment by mm7894215** [comment]
Thanks @Fryuni for the clean multi-profile implementation! Per-profile cursor isolation, backward-compatible default-profile fields, SQL-injection-safe session id quoting, and well-targeted regression tests for the same-second-insert and late-finalization edge cases. Merged — closes #79. Will ship in the next release.
